File: Mobile Programming/Mobile Programming/app/src/main/java/com/example/uasmobileprogramming/RecycleViewAdapter.java
package com.example.uasmobileprogramming;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.view.menu.MenuView;
import androidx.recyclerview.widget.RecyclerView;

import com.example.uasmobileprogramming.model.Data;

import java.util.ArrayList;
import java.util.List;

public class RecycleViewAdapter extends RecyclerView.Adapter<RecycleViewAdapter.ViewHolder> {

    private List<Data> dataList = new ArrayList<>();
    private onViewClick listener;

    @NonNull
    @Override
    public RecycleViewAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_list,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull RecycleViewAdapter.ViewHolder holder, int position) {
        Data data = dataList.get(position);
        holder.tvnama.setText(data.getNama());
        holder.tvharga.setText(data.getHarga());
        holder.tvjumlah.setText(data.getJumlah());
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        TextView tvnama, tvharga, tvjumlah;

        public ViewHolder(@NonNull View itemView){
            super(itemView);
            tvnama = itemView.findViewById(R.id.tvMainNama);
            tvharga = itemView.findViewById(R.id.tvMainHarga);
            tvjumlah = itemView.findViewById(R.id.tvMainJumlah);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    listener.onViewClick(getAdapterPosition());
                }
            });
        }
    }

    public void setDataList(List<Data> list ){
        this.dataList = list;
        notifyDataSetChanged();
    }

    public void clearDataList(List<Data> list ){
        this.dataList = list;
        dataList.clear();
    }

    public interface onViewClick{
        void onViewClick(int position);
    }

    public void setOnClickListener(onViewClick listener){
        this.listener = listener;
    }
}

File: app/src/main/java/com/example/belajar2/RecycleViewAdapter.kt
package com.example.belajar2

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.belajar2.Model.Data
import java.util.ArrayList

class RecycleViewAdapter : RecyclerView.Adapter<RecycleViewAdapter.ViewHolder>() {

    private var dataList: MutableList<Data> = ArrayList<Data>()
    private var listener: onViewClick? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecycleViewAdapter.ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_list, parent, false))
    }

    override fun onBindViewHolder(holder: RecycleViewAdapter.ViewHolder, position: Int) {
        val data = dataList[position]
        holder.tvnama.setText(data.getNama())
        holder.tvharga.setText(data.getHarga())
        holder.tvjumlah.setText(data.getJumlah())
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal var tvnama: TextView
        internal var tvharga: TextView
        internal var tvjumlah: TextView

        init {
            tvnama = itemView.findViewById(R.id.tvMainNama)
            tvharga = itemView.findViewById(R.id.tvMainHarga)
            tvjumlah = itemView.findViewById(R.id.tvMainJumlah)

            itemView.setOnClickListener { listener!!.onViewClick(adapterPosition) }
        }
    }

    fun setDataList(list: MutableList<Data>) {
        this.dataList = list
        notifyDataSetChanged()
    }

    fun clearDataList(list: MutableList<Data>) {
        this.dataList = list
        dataList.clear()
    }

    interface onViewClick {
        fun onViewClick(position: Int)
    }

    fun setOnClickListener(listener: onViewClick) {
        this.listener = listener
    }
}
